spec/daffy_lib/has_encrypted_attributes_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe DaffyLib::HasEncryptedAttributes do
+  describe 'when a subclass does not implement the HasEncryptedAttributes abstract methods' do
+    let(:not_implemented_class) do
+      Class.new(DaffyLib::EncryptionKey) do
+        include DaffyLib::HasEncryptedAttributes
+      end
+    end
+
+    it 'raises a NoMethodError when partition_guid is not implemented' do
+      expect { not_implemented_class.new.generate_partition_guid }.to raise_exception(NoMethodError)
+    end
+
+    it 'raises a NoMethodError when encryption_epoch is not implemented' do
+      expect { not_implemented_class.new.generate_encryption_epoch }.to raise_exception(NoMethodError)
+    end
+  end
+
+  describe 'when a subclass does implement the HasEncryptedAttributes abstract methods' do
+    it 'does not raise a NoMethodError when generate_partition_guid is implemented' do
+      partition_provider = create :partition_provider
+      proxy = create :proxy, partition_provider_guid: partition_provider.guid
+      child = create :child, proxy: proxy
+      expect { child.generate_partition_guid }.not_to raise_error
+    end
+
+    it 'does not raise a NoMethodError when generate_encryption_epoch is implemented' do
+      partition_provider = create :partition_provider
+      proxy = create :proxy, partition_provider_guid: partition_provider.guid
+      child = create :child, proxy: proxy
+      expect { child.generate_encryption_epoch }.not_to raise_error
+    end
+
+    it 'raise a NoMethodError when the partition_guid attribute is changed' do
+      partition_provider = create :partition_provider
+      proxy = create :proxy, partition_provider_guid: partition_provider.guid
+      child = create :child, proxy: proxy
+      child.update(partition_guid: 1234)
+      expect(child.errors).to include(:partition_guid)
+    end
+
+    it 'raise a NoMethodError when the encryption_epoch attribute is changed' do
```

Comment:
This test is just checking that a validation error is raised. It does not check that NoMethodError is being raised. That's not what's being raised.

Reply:
Changed title of test.